web-wallet/src/lib/wallet/__tests__/initializeWallet.spec.js

```diff
@@ -0,0 +1,40 @@
+import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
+import { ProfileGenerator } from "$lib/vendor/w3sper.js/src/mod";
+
+import { settingsStore, walletStore } from "$lib/stores";
+
+import { initializeWallet } from "..";
+
+describe("initializeWallet", () => {
+  const settingsResetSpy = vi
+    .spyOn(settingsStore, "reset")
+    .mockReturnValue(undefined);
+  const clearWalletAndInitSpy = vi
+    .spyOn(walletStore, "clearLocalDataAndInit")
+    .mockResolvedValue(undefined);
+
+  afterEach(() => {
+    settingsResetSpy.mockClear();
+    clearWalletAndInitSpy.mockClear();
+  });
+
+  afterAll(() => {
+    settingsResetSpy.mockRestore();
+    clearWalletAndInitSpy.mockRestore();
+  });
+
+  it("should clear the settings store and initialize a new wallet", async () => {
+    const mnemonic =
+      "cart dad sail wreck robot grit combine noble rap farm slide sad";
```

Comment:
Maybe let's go with

` const mnemonic = "dad dad dad dad dad dad dad dad dad dad dad dad";`

so as to avoid potentially exposing a valid mnemonic.

Reply:
I used that because it's the one you already used in the create flow test, where you had to test with the shuffled mnemonic.

I don't see an issue in having a valid mnemonic in tests though.